Dockerfile

```
@@ -101,4 +101,9 @@ WORKDIR ${RUNNER_PATH}
 
 # COMMAND
 ENV IN_DOCKER=1
-CMD ["cml-runner"]
+# Smart entrypoint understands commands like `bash` or `/bin/sh` but defaults to `cml`;
+# also works for GitLab CI/CD
+# https://gitlab.com/gitlab-org/gitlab-runner/-/blob/4c42e96/shells/bash.go#L18-37
+# https://gitlab.com/gitlab-org/gitlab-runner/-/blob/4c42e96/shells/bash.go#L288
+ENTRYPOINT ["/bin/bash", "-c", "basename -- \"$0\" | grep -qE '^(b?a|z|fi|t?c)?sh$' && exec \"$0\" \"$@\" || exec cml \"$0\" \"$@\""]
```

```suggestion
ENTRYPOINT ["/bin/bash", "-c", "which -s -- \"$0\" && exec \"$0\" \"$@\" || exec cml \"$0\" \"$@\""]
```

@casperdcl, lawful neutral?

---

lawful or neutral evil methinks. Still an upgrade :)